withastro/astro
# unify preview server config

## Changes
Merge preview server config by:
```js
const previewConfig = Object.assign({},
  settings.config.vite.server,
  settings.config.vite.preview,
  settings.config.server,
)
```

Make the preview server can access all server config from `vite.server` / `vite.preview` / `server`.

Especially, this make preview server access `proxy` settings from these objects which allow preview proxy api request just like dev server!

## Testing
1. add proxy config to `astro.config.ts`
2. build your project
3. run `astro preview` and check network





## Docs
This PR adds new feature to `astro preview`, docs should be updated
